lib/plugins/aws/package/compile/events/s3/index.js

```diff
@@ -254,7 +254,9 @@ class AwsCompileS3Events {
 
             iamRoleStatements.push({
               Effect: 'Allow',
-              Resource: `arn:aws:s3:::${bucket}`,
+              Resource: {
+                'Fn::Join': ['', ['arn:', { Ref: 'AWS::Partition' }, `:s3:::${bucket}`]],
```

Comment:
What happens if the `bucket` is not a string but an object (like a `Ref` or an `ImportValue`)?

Reply:
Thanks for the pointer. Yes, we definitely have to support that as well and I saw your PR the other day. I'd say that we ship this PR since it addresses a couple of other blocking issues and after that we should focus on the improvements you've proposed via https://github.com/serverless/serverless/pull/6417

Reply:
thanks!

Reply:
Thanks for working on the fix for that! :+1: